### ISSUE TITLE
Creating a new dataset 

### ISSUE THREAD

**Issue by scientist1642** [CuriousAI/tagger#3]
Hi, I'm trying to create a new unlabeled dataset and had some questions:
https://github.com/CuriousAI/tagger/blob/master/data/shapes.py#L110
As I see masks are only used to calculate AMI score and doesn't take part in training process, right?
What about the "codes", what is it used for? in shapes it's empty and in Freq20-MNIST it seems related to textures.

**Comment by Qwlouse** [comment]
Correct. At some point we've used `codes` to see how well we could classify textures in the image. It is also not used for training. So just using arrays of zeros should work fine.

**Comment by scientist1642** [comment]
Hi @Qwlouse, I am wondering how to choose the rough network size in order to not over/underfit. 
For example, does shapes dataset really need (2000, 1000, 500) ladder?

**Comment by Qwlouse** [comment]
We haven't seen a case of overfitting yet. However for some reason the shapes network needs to be rather big. You can run with fewer units, but performance does degrade.

**Comment by aaronzou** [comment]
Thanks. 

2017年5月11日 星期四 -0700 上午2:54 发件人 Klaus Greff  <notifications@github.com>:

**Comment by aaronzou** [comment]
I got the following print messages when training the model:
  
> Blocks tried to match the sources (['targets_labeled', u'codes', 'masks_labeled', 'features_labeled', 'features_unlabeled', 'masks_unlabeled']) of the training dataset to the names of the Theano variables (['features_unlabeled']), but failed to do so. If you want to train on a subset of the sources that your dataset provides, pass the `sources` keyword argument to its constructor. Or pass on_unused_sources='warn' or on_unused_sources='ignore' to the GradientDescent algorithm.
> Monitoring on auxiliary data started
Is this normal? 

After that It seems starting epochs. 

> _e 3, i 1500:
> 	Train_Denoising_Cost 0.279 0.221 0.213
> 	Validation_Denoising_Cost 0.262 0.205 0.196
> 	rd 1.46 s tr 1492.44 s_ 

**But when I run  nvidia-smi on another terminal.  I cannot see this process using the GPU? Is that normal ? Doest it mean I am training the model on CPU?  Why is GPU not used?**

**Comment by aaronzou** [comment]
@Qwlouse Hi,
I find all  your experimental examples of shapes contain constantly 3 groups.  
Could I use your framework to handle the examples whose label count varies from 3~8? 
 A simple way I can figure out is to separately train 6 batches: 3-label examples , 4-label examples, 5-label examples, 6-label examples , 7-label examples, 8-label examples.  
Could I train all those examples together?  


Another question is that our examples were generated from more than 500 polygons (yours is 2: triangles and squares) and the group number of each example varies from 3 to 8. 
Do you think how many training examples I need ?
